vpr/src/route/rr_graph2.cpp

```diff
@@ -1117,8 +1179,10 @@ static void load_block_rr_indices(const DeviceGrid& grid,
                     for (int height_offset = 0; height_offset < type->height; ++height_offset) {
                         int y_tile = y + height_offset;
                         for (e_side side : SIDES) {
-                            VTR_ASSERT(indices[IPIN][x_tile][y_tile][side].size() == size_t(type->num_pins));
-                            VTR_ASSERT(indices[OPIN][x_tile][y_tile][side].size() == size_t(type->num_pins));
+                            VTR_ASSERT((indices[IPIN][x_tile][y_tile][side].size() == size_t(type->num_pins))
```

Comment:
I don't understand what these asserts are checking.  I think we can now have any number of input pins at a side and offset location between 0 and type->num_pins.  But these asserts are checking for either all pins or no pins are on a given side.

Reply:
Hi Vaughn,
    The rr_node_indices has a fixed-length vector for each side of a grid. If it is zero, it means there are no pins on this side. Otherwise, its length will be the number of pins, because client functions will send data query to get the rr_node id for a given grid at a given side with a pin index (ptc_num). If the node does not exist on the location, the indices will just store an OPEN id. As a result, the length is fixed for the number of pins where you can see a lot of OPEN node ids are inserted.
I have commented in the codes now and hope it will help other guys to understand how it is  organized

Reply:
OK, thanks.  Yikes, that seems like a very inefficient way to store this data, even temporarily.  But that's another issue:  I'll file one to take a look at that.

Reply:
Did some back of the envelope calculations, and I think the data structure would be (very roughly) ~150 MB for a big Titan design.  Significant, but may not be worth recoding as an unordered map, so I didn't file the other issue for now.